dask_cloudprovider/providers/aws/ecs.py

```diff
@@ -135,14 +137,7 @@ async def _():
 
     @property
     def _use_public_ip(self):
-        # Fargate needs public IP for image pull, EC2 doesn't support public IP, therefore
-        # we will assume for now that we will use a public IP when in Fargate mode and not
-        # when in EC2 mode.
-
-        # TODO Fargate can also use a NAT to pull the image so we could allow this to be false
-        # when in Fargate provided there is a NAT
-
-        return self.fargate
+        return not self._use_private_ip
```

Comment:
should be `self.fargate and not self._use_private_ip`, no? Or otherwise, should be clear what are the right defaults for ec2 versus fargate (perhaps in init rather than here).

Reply:
Thanks, I agree with that. Updated to the logic you provided.